Fix wrong device JDEC ID

CreativeRobotics opened the pull request:
Fixed an incorrect device memory_type value for W25Q80DL.
Added two new devices W25Q80DV and MX25L6433F.
Added example flash_info_custom_device, derived from the flash_info example that demonstrates how to define and use devices not already listed in flash_devices.h

hathach reviewed (commented):
Thank you for your PR, it look good, though I think having dedicated example for custom info is not needed. And everyone will just have their own custom flash. Could you modify the existing flash_info example. And have the `static const SPIFlash_Device_t my_flash_devices[]` commented by default, with text to tell where people to find existing flash_devices.h etc ...

CreativeRobotics commented:
Ok, I deleted the new example and updated the flash_info example with commented out code and some explanation of how to define and deploy other memory devices.

hathach reviewed (approved):
superb !! The comment text is much more than I could expect, great work. will merge when Ci passed.
